modules/cloudbuild/main.tf

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cloudbuild_project_id       = format("%s-%s", var.project_prefix, "cloudbuild")
+  cloudbuild_apis             = ["cloudbuild.googleapis.com", "sourcerepo.googleapis.com", "cloudkms.googleapis.com"]
+  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
+  activate_apis               = distinct(var.activate_apis)
+}
+
+resource "random_id" "suffix" {
+  byte_length = 2
+}
+
+data "google_organization" "org" {
+  organization = var.organization_id
+}
+
+
+/******************************************
+  Cloudbuild project
+*******************************************/
+
+module "cloudbuild_project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 5.0"
+  name                        = local.cloudbuild_project_id
+  random_project_id           = "true"
+  disable_services_on_destroy = "false"
+  folder_id                   = var.folder_id
+  org_id                      = var.organization_id
+  billing_account             = var.billing_account
+  activate_apis               = local.activate_apis
+}
+
+resource "google_project_service" "cloudbuild_apis" {
```

Comment:
It should be possible to move these in to the Project Factory module and enforce dependencies through the `project_id` output as it blocks on the service resources.

Reply:
Yeah agree, although my experience has been that there are occasional race conditions using this approach. So it is a trade off between stability vs how concise / clean the code is. I can switch back but will likely cause intermittent failures in tests and also in wild?

Reply:
The two approaches should be functionally equivalent. I would like to understand why if they aren't but we don't need to block on this.